### ISSUE TITLE
Test .RPMDistributorTestCase.test_update_checksum_type contains a test teardown failure.

### ISSUE THREAD

**Issue by kersommoura** [PulpQE/pulp-smash#762]
Analysing the Build #199 (Aug 19, 2017 5:34:16 AM) from Jenkins, it was reported that the aforementioned test failed due to a time out.

Part of Stacktrace

```
cls = <class 'pulp_smash.tests.rpm.api_v2.test_crud.RPMDistributorTestCase'>

    @classmethod
    def tearDownClass(cls):
        """Delete all resources named by ``resources``."""
        client = api.Client(cls.cfg)
        for resource in cls.resources:
            client.delete(resource)
>       client.delete(ORPHANS_PATH)

 pulp_smash.exceptions.TaskTimedOutError: Task /pulp/api/v2/tasks/587ad896-3332-4201-b296-c67c8fd4fd8f/ is ongoing after 360 polls.


../../.virtualenvs/pulp-smash/src/pulp-smash/pulp_smash/api.py:411: TaskTimedOutError
```
It took 33 min to run the test.

I ran again the test against the following system:
-   Platform Version: 2.13.4a1
-   Red Hat Enterprise Linux Server release 7.4 (Maipo)

Once again the test failed, and it took more than 30 minutes to ran.

Since an exception is raised during the execution of teardown, a possible solution is to use addCleanup method to execute the final cleaning during the test execution.

**Comment by omaciel** [comment]
@kersommoura can you take a look at the system's logs as it pertains to `Pulp` to see if there's anything interesting there? Also, if you refactor the test to use `addCleanup` and re-execute the test manually, does it pass? I'm wondering if there is something else about the test that may be generating a valid failure for when you attempt to delete the artifact from the system...

**Comment by Ichimonji10** [comment]
@omaciel Sometimes, Pulp just breaks. The breakage is infrequent and random. I'm not 100% sure why it occurs, but I strongly suspect that it's related to the use of mongodb, which lacks consistency guarantees.

If one looks over the Jenkins test results over a period of time, you'll notice that these random failures manifest themselves in a variety of ways. Sometimes, an HTTP 500 error will be thrown with a message about duplicate primary keys. Other times, a task will time out for no discernible reason. Once a test fails, if that same test is run on the same host again, then that test is especially likely to fail again. And again, and again, and again. If one resets Pulp (by calling `python -c 'from pulp_smash import config, utils; utils.reset_pulp(config.get_config())`), then that test is likely to start *succeeding* again.

If one looks over the Jenkins test results over a period of time, you'll notice that these random failures are especially likely to manifest themselves as calls to `client.delete(ORPHANS_PATH)`. I'm not sure why this is. It may be because the call is especially good at producing errors in Pulp. It may be because this line of code is called very frequently.

There's nothing wrong with calling `client.delete(ORPHANS_PATH)`. It's a well-documented and stable interface in Pulp 2. And there's nothing wrong with the tear-down logic listed in this issue. The logic is sane. Pulp just breaks sometimes.

There's nothing we can do to get rid of these types of errors completely in Pulp 2. I don't think we can get rid of them in Pulp 2. The best we can do is to try to mitigate the impact of these random failures. I think a good way to do this is to make fewer calls to `client.delete(ORPHANS_PATH)`. We don't need most of the calls, anyway. They just blow away Pulp's cache, which makes test runs take longer. We can accomplish this by walking through test cases that inherit from `pulp_smash.utils.BaseAPITestCase` and refactoring them to inherit from `unittest.TestCase` instead. As a nice side-effect, we'll get more robust clean-up logic too.

**Comment by kersommoura** [comment]
@omaciel I did execute the code again, generate a log as well, and took a look at the system's logs.  It took close to 35 minutes to run the test again. As @Ichimonji10 mentioned I reset pulp and the test execution was fast, 11 seconds, without failures, or time-out.

**Comment by Ichimonji10** [comment]
We've discussed this and think that this issue can be closed in favor of #763.
